src/server/modemwatcher.cpp

```diff
@@ -31,15 +33,37 @@ ModemWatcher::ModemWatcher(const QString objectPath, const QString interface, QO
     QObject(parent), m_objectPath(objectPath), m_interface(interface), m_interfaceAvailable(false)
 {
     QDBusConnection::systemBus().connect(OfonoConstants::OFONO_SERVICE, m_objectPath,
-                                         OfonoConstants::OFONO_MODEM_INTERFACE, "PropertyChanged",
-                                         this, SLOT(onModemPropertyChanged(QString, QDBusVariant)));
+                                         OfonoConstants::OFONO_MODEM_INTERFACE,
+                                         "PropertyChanged", this,
+                                         SLOT(onModemPropertyChanged(QString, QDBusVariant)));
 
+    m_ofonoWatcher = new QDBusServiceWatcher(OfonoConstants::OFONO_SERVICE,
+                                             QDBusConnection::systemBus(),
+                                             QDBusServiceWatcher::WatchForRegistration,
+                                             this);
+    connect(m_ofonoWatcher, SIGNAL(serviceRegistered(QString)),
+            this, SLOT(getProperties()));
+
+    if (QDBusConnection::systemBus().interface()->isServiceRegistered(OfonoConstants::OFONO_SERVICE))
```

Comment:
I am not aware of a trivial way to communicate asynchronously with the dbus daemon (via Qt APIs). As you said it is unlikely to block, or at least unlikely to block for so long that timed operations or device operations running timed would be affected.